controllers/utils.go

```diff
@@ -70,15 +68,26 @@ func resolveMondooOperatorImage(log logr.Logger, userImageName, userImageTag str
 	}
 	mondooContainer := useImage + ":" + useTag
 
-	imageUrl, err := parseReference(log, mondooContainer)
-
+	imageUrl, err := skipResolveImage(resolveImage, log, mondooContainer)
 	if err != nil {
-		log.Error(err, "Failed to parse reference")
-		return "", err
+		log.Error(err, "Failed to skip resolving image")
+		return imageUrl, err
 	}
 	return imageUrl, nil
 }
 
+func skipResolveImage(resolveImage bool, log logr.Logger, mondooContainer string) (string, error) {
+	if !resolveImage {
+		imageUrl, err := parseReference(log, mondooContainer)
+		if err != nil {
+			log.Error(err, "Failed to parse reference")
+			return "", err
+		}
+		return imageUrl, nil
+	}
+	return mondooContainer, nil
+}
+
```

Comment:
The naming here should make it clear what the function is doing (returning the image to use for the Pod), and the boolean that controls whether or not to skip the image-to-SHA resolution should indicate its purpose.
 
```suggestion
func getImage(skipResolveImage bool, log logr.Logger, mondooContainer string) (string, error) {
	if !skipResolveImage {
		imageUrl, err := parseReference(log, mondooContainer)
		if err != nil {
			log.Error(err, "Failed to parse reference")
			return "", err
		}
		return imageUrl, nil
	}
	return mondooContainer, nil
}

```